### DISCUSSION TITLE
Can I configure Velvet?

### DISCUSSION THREAD

**Discussion by Chessisbom** [mhonert/velvet-chess#11] [Q&A]
Hi! I would like to know if I can configure the bot's aggression, can I make it more or less agressive?

Thanks,

Chess Is Bom

**Comment by mhonert** [comment]
Hi!

Currently there are no configuration options to make it play more or less aggressive.
The evaluation uses a neural network, so there are no evaluation terms, that could be made configurable.
There are certain internal search parameters, that could have an effect on playing style, but it would require some experimentation
to see, whether this produces something interesting. An alternative could be to train different networks.

I cannot promise anything at the moment, but will put the topic onto my todo list for further investigation.
What is already on the todo list is the implementation of different strength levels.

Best regards

**Reply by Chessisbom** [reply]
Thanks! I was searching for an aggressive as sharp as rashid nezhmeditinov. I will keep waiting. Keep up with the good work, cheers!

**Comment by tissatussa** [comment]
> ..make it play more or less aggressive

i wonder what 'aggressive' means .. is it playing moves which aim to keep threats and can only be defended accurately ? That way 'aggressive' also means taking a risk : when the opponent DOES find adequate defensive moves, a resulting position will NOT be advantageous ?

> The evaluation uses a neural network, so there are no evaluation terms, that could be made configurable. There are certain internal search parameters, that could have an effect on playing style, but it would require some experimentation to see, whether this produces something interesting.

but the search function will find SOME moves to be evaluated by the network, so the search function can be configurable ? I mean : are ALL possible moves / variations in a position evaluated ? I guess not ..

**Comment by mhonert** [comment]
I just released v8.0.0, which supports the UCI "Style" option to choose between a "Normal" (default) and a "Risky" style.
The "risky" style increases the chance for "aggressive" play, but comes with the downside, that it reduces the overall strength by around 25 Elo (there is the risk that a failed attack leads into an unfavorable position).
